### PR TITLE
Add Infinix HOT 2 X510 support

### PR DESCRIPTION
**Pull request by Nonta72**
Compiled and tested
Working fine.

### CODE REVIEW

**Comment by Nonta72** [comment]
Sorry for the first pull request!
Had to re-fork the base repo and start over.

File: libmbp/devices/google.cpp [hunk 1]
```diff
@@ -40,6 +40,32 @@ void addGoogleDevices(std::vector<Device *> *devices)
...
+        device->setSystemBlockDevs({ MTK_SYSTEM, "/dev/block/mmcblk0p23" });
+        device->setCacheBlockDevs({ MTK_CACHE, "/dev/block/mmcblk0p24" });
+        device->setDataBlockDevs({ MTK_USERDATA, "/dev/block/mmcblk0p25" });
+        device->setBootBlockDevs({ MTK_BOOT, "/dev/block/mmcblk0p7" });
+        device->setRecoveryBlockDevs({ MTK_RECOVERY, "/dev/block/mmcblk0p8" });
+        devices->push_back(device);
+        
+    // Infinix X510 (Infinix ROM partition scheme)
+        device = new Device();
+        device->setId("d5110_infinix");
```
Comment (Caio99BR) [inline libmbp/devices/google.cpp]: You cant have same ID two times

**Comment by Nonta72** [comment]
Well,
I have compiled a debug build without issue.
The problem is that this device comes in different versions. One with Android One and the other one without Android One. Partition schemes are different. While Android One ROM have system on block 23, Infinix ROM have it on block 18 as you can see.

File: libmbp/devices/google.cpp [hunk 2]
```diff
@@ -40,6 +40,32 @@ void addGoogleDevices(std::vector<Device *> *devices)
     device->setBootBlockDevs({ MTK_BOOT, "/dev/block/mmcblk0p7" });
     device->setRecoveryBlockDevs({ MTK_RECOVERY, "/dev/block/mmcblk0p8" });
     devices->push_back(device);
+    
+    // Infinix X510 (Android One)
+        device = new Device();
+        device->setId("d5110_infinix");
```
Comment (hankching) [inline libmbp/devices/google.cpp]: Suggest as follows:
device->setId("X510_AndroidOne");

Comment (hankching) [inline libmbp/devices/google.cpp]: Another use as follows:
device->setId("X510_scheme");

Comment (Nonta72) [inline libmbp/devices/google.cpp]: Testing now with those changes.

**Comment by Nonta72** [comment]
An error!
[ 93%] Create boot UI archive (armeabi-v7a)
CMake Error: Unknown option to -E tar: --format=zip
make[2]: *** [android/CMakeFiles/bootui-archive_armeabi-v7a] Error 1
make[1]: *** [android/CMakeFiles/bootui-archive_armeabi-v7a.dir/all] Error 2
make: *** [all] Error 2
nonta72@HaswellGT2:~/DualBootPatcher/build$

**Comment by hankching** [comment]
Try this first

device->twOptions()->supported = false;

**Comment by Nonta72** [comment]
> device->twOptions()->supported = false;

In which file ?

And I think that the followings fit better in my situation:
For Android One: `device->setId("d5110");`
For Infinix ROM: `device->setId("d5110_infinix");`

**Comment by hankching** [comment]
ID for identify which one is which only, so that is OK.

Which one? rebuild or boot stage.

**Comment by Nonta72** [comment]
It's fixed.
I will close this pull and re-open another one with changes recently made.
Thank you.

**Comment by Caio99BR** [comment]
@chenxiaolong why not add to Milestone 9?

**Comment by chenxiaolong** [comment]
What are the `MTK_11120000_*` paths for? They aren't used by either device definition.

**Comment by Nonta72** [comment]
The mounting method has changed in Android 6.0.
In LP kernel 3.10, it is dev/block/platform/mtk-msdc.0/by-name
But in MM kernel 3.18, it is dev/block/platform/mtk-msdc.0**/11120000**/by-name
If we don't use that, the app might fail to identify boot partition in android 6.0 for HOT 2.
Should I add another device definition with these paths ?

**Comment by hankching** [comment]
You don't understand the following sentence:
>They aren't used by either device definition.

Which means it must see inside google.cpp file.

**Comment by Nonta72** [comment]
I have sold my device.
Anyone feel free to make another pull request.
I might reopen it if I get the device back.

**Pull request closed without merging**